### PR TITLE
Fixes #25242 - Add a lockfile for openscap spool

### PR DESCRIPTION
**Pull request by xprazak2**

### CODE REVIEW

**Comment by xprazak2** [comment]
@lzap, this should prevent execution of the script when the previous one is still running

File: lib/smart_proxy_openscap/openscap_lib.rb [hunk 1]
```diff
@@ -57,7 +57,14 @@ def self.common_name(request)
...
     arf_dir = File.join(loaded_settings.spooldir, "/arf")
     return unless File.exist? arf_dir
+    lockfile = File.join(loaded_settings.spooldir, ".spool.lock")
+    if File.exist? lockfile
+      logger.debug "Lock file #{lockfile} for openscap spool exists, not sending spool to server"
+      return
+    end
+    FileUtils.touch lockfile
     SpoolForwarder.new(loaded_settings).post_arf_from_spool(arf_dir)
+    FileUtils.rm lockfile
```
Comment (lzap) [inline lib/smart_proxy_openscap/openscap_lib.rb]: Can you put the removal to `ensure` block? We should probably also install SIGTERM trap to remove it as well.

**Comment by xprazak2** [comment]
Updated, I moved the changes to binary, added ensure block and sigterm trap.

File: bin/smart-proxy-openscap-send [hunk 2]
```diff
@@ -37,10 +37,25 @@ if !Proxy::SETTINGS.foreman_url
...
   puts "#{e} See #{Proxy::OpenSCAP.fullpath(loaded_settings.openscap_send_log_file)}"
   exit 438
+ensure
+  FileUtils.rm lockfile
 end
+
+Signal.trap("TERM") {
+  FileUtils.rm lockfile
+  exit
+}
```
Comment (lzap) [inline bin/smart-proxy-openscap-send]: This block must be moved just under line 40, this way it is never executed so the trap will not be active during execution.

Additionally, lockfile can be not yet created, so please check if it exists and delete it.

**Comment by xprazak2** [comment]
I moved the trap and changed lock file not to be hidden.

**Comment by lzap** [comment]
@ares can you merge this for us? I tested this, works fine. 6.4 bug.

**Comment by ares** [comment]
Thanks @xprazak2 for quick merge and @lzap for prompt testing. Merging and will do the release 0.7.1 soon after we got the second PR in (timeout configuration)

**Pull request merged**